### PR TITLE
Convert most shared_ptr<DiscretisedDensity> arguments to shared_ptr<const...>

### PR DESCRIPTION
**Pull request by KrisThielemans**
This increases safety, similar to #470

### CODE REVIEW

**Comment by KrisThielemans** [comment]
@rijobro this should handle most DiscretisedDensity case. It might need a SIRF change as well. Would you be able to check?

Did you have a look at `exam_info`? If not, probably easiest to add it on top of this one.

**Comment by KrisThielemans** [comment]
Travis is ok

**Comment by rijobro** [comment]
Will have a look this morning. Also need to check that it compiles in Debug mode, to avoid needing a repeat of #483.

**Comment by rijobro** [comment]
Should perhaps think about having a travis in debug mode.

**Comment by KrisThielemans** [comment]
I've done most of the exam-info. Was sadly more work than I hoped. I will push this soon

File: src/Shape_buildblock/DiscretisedShape3D.cxx [hunk 1]
```diff
@@ -127,10 +127,10 @@ DiscretisedShape3D(const VoxelsOnCartesianGrid<float>& image_v)
 
 
 DiscretisedShape3D::
-DiscretisedShape3D(const shared_ptr<DiscretisedDensity<3,float> >& density_ptr_v)
-   : density_ptr(density_ptr_v)
+DiscretisedShape3D(const shared_ptr<const DiscretisedDensity<3,float> >& density_ptr_v)
```
Comment (rijobro) [inline src/Shape_buildblock/DiscretisedShape3D.cxx:130]: change variable names to sptr?

**Comment by rijobro** [comment]
Ok thanks. So far everything builds on debug, too. In terms of review, didn't see anything out of the ordinary, but there was too much to look at in detail.

**Comment by rijobro** [comment]
From SIRF:
```
auto sptr_im = prior.get_anatomical_image_sptr();
shared_ptr<STIRImageData> sptr_id(new STIRImageData(sptr_im));
return newObjectHandle(sptr_id);
```
Constructor of STIRImageData doesn't take a const DiscDensity as it needs to be able to modify the data if desired. This means lots of copying of data, but unavoidable I think.

**Comment by KrisThielemans** [comment]
SIRF PR is https://github.com/CCPPETMR/SIRF/pull/615

**Pull request merged**